camayoc/tests/qcs/api/credentials/test_creds_common.py

```diff
@@ -0,0 +1,257 @@
+    :steps: Send POST with necessary data to documented api endpoint.
+    :expectedresults: Error is thrown and no new credential is created.
+    """
+    client = api.Client(api.echo_handler)
+    cred = Credential(
+        cred_type=src_type,
+        client=client,
+        username='',
+        password=uuid4(),
+    )
+    response = cred.create()
+    assert response.status_code == 400
+    assert cred._id is None
+
+
+@pytest.mark.parametrize('src_type', QCS_SOURCE_TYPES)
+def test_negative_create_missing(src_type, cleanup):
+    """Attempt to create a credential missing a password.
+
+    The request should be met with a 4XX response.
+
+    :id: 97a24094-3e9b-4eca-884e-3eda4e461ea1
+    :description: Create a credential missing a password.
+    :steps: Send POST with necessary data to documented api endpoint.
+    :expectedresults: Error is thrown and no new credential is created.
+    """
+    client = api.Client(api.echo_handler)
+    cred = Credential(cred_type=src_type, client=client)
+    response = cred.create()
+    assert response.status_code == 400
+    assert cred._id is None
+
+
+@pytest.mark.parametrize('src_type', QCS_SOURCE_TYPES)
+def test_read_all(src_type, shared_client, cleanup):
+    """After created, retrieve all credentials with GET to api.
+
+    :id: fa05b857-5b01-4388-9226-8dfb5639c815
+    :description: The API should return list with all credentials created
+        when a GET request is sent to the credential endpoint.
+    :steps:
+        1) Create collection of credentials, saving the information.
+        2) Send GET request to credential endpoint to get list of
+           created credentials.
+        3) Confirm that all creds are in the list.
+    :expectedresults: All creds are present in data returned by API.
+    """
+    creds = []
+    for _ in range(random.randint(2, 5)):
+        cred = Credential(
+            cred_type=src_type,
+            client=shared_client,
+            password=uuid4())
+        cred.create()
+        # add the credential to the list to destroy after the test is done
+        cleanup.append(cred)
+        assert_matches_server(cred)
+        creds.append(cred)
+
+    remote_creds = Credential().list().json()
+    for local, remote in zip(creds, remote_creds):
+        local.equivalent(remote)
+
+
+@pytest.mark.parametrize('src_type', QCS_SOURCE_TYPES)
+def test_read_indv(src_type, shared_client, cleanup):
+    """After created, retrieve each credential with GET to api.
+
+    :id: 4d381119-2dc3-42b6-9b41-e27307d61fcc
+    :description: The API should a single credential when a GET is made
+        to the credentials path and a id is specified.
+    :steps:
+        1) Create collection of credentials, saving the information.
+        2) Send GET request to credential endpoint with the id
+           specified.
+        3) Confirm that each is retrieved
+    :expectedresults: All creds are present in data returned by API.
```

Comment:
You are reading individual credentials not all credentials at once. That said I think this can be reworded to say something like "Each credential can be read and has proper data returned."